hw_photosort/photosort.py

```diff
@@ -0,0 +1,153 @@
+import os
+import PIL
+from PIL import Image
+import shutil
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("input_dir", help="input directory")
+parser.add_argument("output_dir", help="output directory")
+parser.add_argument("-m", "--map", help="generate html", action="store_true")
+parser.add_argument("-x", "--delete", help="delete photos", action="store_true")
+parser.add_argument("-r", "--recursively", help="process directories recursively", action="store_true")
+args = parser.parse_args()
+
+directory = args.input_dir
+output_directory = args.output_dir
+photos = []
+delete_image = args.delete
+generate_html = args.map
+recursively = args.recursively
+
+class Photo:
+    def __init__ (self, path, name, date, year, latitude, longitude):
+        self.path = path
+        self.name = name
+        self.date = date
+        self.year = year
+        self.latitude = latitude
+        self.longitude = longitude
+
+    def __str__(self):
+        return "{0}, {1}, {2}, {3}, {4}, {5}".format(self.path, self.name, self.date, self.year, self.latitude, self.longitude)
+
+
+def open_directory(directory):
+    print("directory: {0}".format(directory))
+    list_of_photos = os.listdir(directory)   
+    for photo in list_of_photos:
+        path = os.path.join(directory, photo)
+        if os.path.isdir(path):
+            if(recursively):
+                open_directory(path)
+            else:
+                print("ignore: {0}".format(path))
```

Comment:
SUPER! Komunikace s uzivatelem je vzdy potreba, aby vedel proc neco nefunguje tak jak si on predstavuje. Idealni by bylo po serii ignored i napsat proc, napr. ze zpracovavame obrazky s koncovkou presne `jpg`. Jinak se to dozvim jen po precteni zdrojoveho kodu.